src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -332,8 +360,31 @@ public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, Create
                 throw new NotImplementedException();
 
             object? wrapperMaybeLocal = wrapperMaybe;
-            retValue = null;
-            throw new NotImplementedException();
+            retValue = impl.CreateObject(externalComObject, flags);
+            if (retValue == null)
+            {
+                // If ComWrappers instance cannot create wrapper, we can do nothing here.
+                return false;
+            }
+
+            if (flags.HasFlag(CreateObjectFlags.UniqueInstance))
+            {
+                // No need to cache NativeObjectWrapper for unique instances. They are not cached.
+                return true;
+            }
+
+            NativeObjectWrapper wrapper;
+            if (impl._rcwTable.TryGetValue(retValue, out wrapper))
+            {
+                return true;
+            }
+
+            wrapper = impl._rcwTable.GetValue(retValue, (c) =>
+            {
+                return new NativeObjectWrapper(externalComObject);
+            });
+            wrapper.AddRef();
```

Comment:
Would it make more sense to have this AddRef in constructor since the matching Release is in finalizer?